public/app/features/query/state/runRequest.ts

```diff
@@ -176,7 +176,7 @@ export function callQueryMethod(
 ) {
   // If any query has an expression, use the expression endpoint
   for (const target of request.targets) {
-    if (isExpressionReference(target.datasource)) {
+    if (isExpressionReference(target.datasource) && request.publicDashboardAccessToken === '') {
```

Comment:
why is this necessary?  can the normal path just handle things?

Reply:
This would return an ExpressionDatasource, which would end up sending queries to the wrong endpoint. Since the public dashboard queries get gathered and ran on the backend, we can skip this.

Reply:
@owensmallwood is it feasible to handle rerouting after the ExpressionDatasource is returned? While we're validating 99%(for now) case here that will change over time.

Reply:
@ryantxu Made a small change to this so that's its a bit clearer. But the context around this is that a panel on a public dashboard will only send a single request to the public dashboard api route which will parse the queries from the dashboard definition that exists in the db, and execute them on the backend. So for this case, if its a public dashboard datasource, we just directly return the result.